exporter/prometheusremotewriteexporter/config.go

```diff
@@ -79,24 +79,29 @@ type RemoteWriteQueue struct {
 	NumConsumers int `mapstructure:"num_consumers"`
 }
 
-// TODO(jbd): Add capacity, max_samples_per_send to QueueConfig.
-
-var _ component.Config = (*Config)(nil)
-
-// Validate checks if the exporter configuration is valid
-func (cfg *Config) Validate() error {
-	if cfg.RemoteWriteQueue.QueueSize < 0 {
-		return fmt.Errorf("remote write queue size can't be negative")
+func (cfg *RemoteWriteQueue) Validate() error {
```

Comment:
I see the tests pass, but I don't see where this is called from.  Am I missing something?

Reply:
Though the comment is not that accurate. https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/config.go#L46 we call "Validate()" on all members of the Config, this is because we want for example fi you include configs from `confighttp` or `exporterhelper` to not have to call Validate because you will forget, or if we implement that method you will not update your code to call it.

May worth better document this and explain to everyone.